core/arch/arm/kernel/thread_spmc_a32.S

```diff
@@ -50,8 +50,10 @@ END_FUNC ffa_msg_send_direct_resp
 FUNC thread_std_smc_entry , :
 UNWIND(	.cantunwind)
 
+	push	{r4, r5} /* Pass these following the arm32 calling convention */
 	ror	r4, r0, #16 /* Save target info with src and dst swapped */
```

Comment:
Shouldn't this be done before pushing `r4`?

Reply:
No, it is the content of `r4` when `thread_std_smc_entry()` was called we'd like to pass on.
What we do with `r4` after the push has nothing to do with that, `r4` is just another preserved register.

Reply:
Of course :facepalm: